src/releaseProvider.ts

```diff
@@ -0,0 +1,77 @@
+import { env, ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+import * as semver from "semver";
+
+const versionKey = "metals-server-version";
+
+export class ReleaseProvider {
+  private readonly state: typeof this.context.globalState;
+
+  constructor(
+    private readonly context: ExtensionContext,
+    // TODO switch to extension version maybe?
```

Comment:
I think server version is better in this case, but we should limit it to full versions, not snapshots.